components/ws-daemon/pkg/container/containerd.go

```diff
@@ -476,9 +480,31 @@ func (s *Containerd) ContainerPID(ctx context.Context, id ID) (pid uint64, err e
 	return uint64(info.PID), nil
 }
 
-// ContainerPID returns the PID of the container's namespace root process, e.g. the container shim.
 func (s *Containerd) IsContainerdReady(ctx context.Context) (bool, error) {
-	return s.Client.IsServing(ctx)
+	if len(s.registryFacadeHost) == 0 {
+		return s.Client.IsServing(ctx)
+	}
+
+	// check registry facade can reach containerd and returns image not found.
+	isServing, err := s.Client.IsServing(ctx)
+	if err != nil {
+		return false, err
+	}
+
+	if !isServing {
+		return false, nil
+	}
+
+	_, err = s.Client.GetImage(ctx, fmt.Sprintf("%v/not-a-valid-image:latest", s.registryFacadeHost))
+	if err != nil {
+		if errdefs.IsNotFound(err) {
+			return true, nil
+		}
+
+		return false, nil
+	}
+
+	return false, nil
```

Comment:
In the unlikely case that the GetImage call succeeded we'd be ready, too, no? I.e. shouldn't this be

```suggestion
	return true, nil
```

Reply:
That should not happen, we should always get not found